### PR TITLE
Chore: Set dimensions on parent element, Preserve graph container contents

### PR DESCRIPTION
**Pull request by cizl**
* Set dimensions on parent element when undefined (https://github.com/memgraph/orb/issues/23)
* Preserve graph container children (https://github.com/memgraph/orb/issues/22)

### CODE REVIEW

File: src/utils/html.utils.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+export const setupContainer = (container: HTMLElement) => {
+  container.style.position = 'relative';
+  const style = getComputedStyle(container);
+  if (!style.display) {
+    container.style.display = 'block';
+    console.warn("[Orb] Graph container doesn't have defined 'display' property. Setting 'display' to 'block'...");
+  }
+  if (!style.width || style.width === '0px') {
```
Comment (tonilastre) [inline src/utils/html.utils.ts:8]: Is this covering cases like `'0rem'`, `'0'`, `'0 px'` if those are possible values for `.width` and `.height` (down below)?

File: src/views/default-view.ts [hunk 2]
```diff
@@ -389,6 +388,17 @@ export class DefaultView<N extends INodeBase, E extends IEdgeBase> implements IO
     }
   };
 
+  private _initCanvas(): HTMLCanvasElement {
+    const canvas = document.createElement('canvas');
+    canvas.id = 'orbCanvas';
```
Comment (tonilastre) [inline src/views/default-view.ts:393]: What will happen if there is a `HTMLElement` with the same `id` in the application using Orb? Do we want to introduce `uuid` for these kind of ids?

Comment (cizl) [reply src/views/default-view.ts:393]: Good comment, I was just wondering that. I'm assigning `uuid`s in `ngx-orb`, but we may want to have those in here as well. But I'm not sure wether this is the responsibility of the user or the library. It also makes sense to me that the user takes care of his own container and orb instances.

This is effectively a leftover so I'll delete it. I used this to get the canvas element but then i realized that I already have a reference to the element (`this._canvas`)

File: src/utils/html.utils.ts (hunk 1 above)
Comment (cizl) [reply src/utils/html.utils.ts:8]: Good point. In firefox the `getComputedSize` returns a `0px` specifically. I can check if it `startsWith('0')` or something like that.

Another point is, what if the user programmatically is switching sizes, then we override that with this condition.

Comment (tonilastre) [reply src/utils/html.utils.ts:8]: Yep, that should do the trick. I just tried this:

```html
<div id="testing-height" style="height: 010px"></div>
```

```js
document.getElementById("testing-height").style.height // outputs '10px'
```

All in all, `010px` is a valid style, but luckily browser will take out the leading zeros.

Regarding user switching sizes, could we maybe then add settings param for it? Which is enabled by default. What do you think?

Comment (cizl) [reply src/utils/html.utils.ts]: @tonilastre That definitely makes sense! Justin Case - I added a regex to capture all "collapsed" style dimensions.

I've been meaning to talk to you about our settings. I'm not very happy with the current implementation. It's tricky to consume settings from a DX perspective. Plus, now they have duplicated fields. We could add a base interface for ViewSettings in the future.

Comment (tonilastre) [reply src/utils/html.utils.ts]: I agree with you regarding settings. I think we should do that along with the following fix: https://github.com/memgraph/orb/issues/16. With that in mind, we could create a baseline interface as well as `AbstractView`.

File: src/utils/html.utils.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        'please make sure that the parent element of the graph container has a defined position and height.',
+        "Setting the height of the graph container to an arbitrary value of '400px'...",
+      );
+    } else {
+      console.warn("[Orb] Graph container doesn't have defined height. Setting height to 100%...");
+    }
+  }
+};
+
+export const collapsedDimensionRegex = /^\s*0+\s*(?:px|rem|em|vh|vw)?\s*$/;
```
Comment (tonilastre) [inline src/utils/html.utils.ts:42]: If `px`, `rem`, `em` can be the upper case then you need to add an ignore-case flag at the end of the regex, `/..../i;`

**Review by tonilastre — APPROVED** [review]

**Pull request merged**